### DISCUSSION TITLE
Dag run status and which repetitive dags/tasks failed for week

### DISCUSSION THREAD

**Discussion by Praful19** [apache/airflow#21931] [Q&A]
HI Team,

I have requirement which is as below.

1. find Dag run status: like how much time dags took time to complete. start ,end and execution time of Dag. 
2. How to get list of repetitively failed dags/tasks for week or month. if we have 150+ dags.  My All dags are running by 12/7.
3. How to generate report for the same mentioned above

Any suggestion please

**Comment by potiuk** [comment]
You can use the REST API for that. https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html
